stream-chat-android-ui-common/src/test/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListViewModelTest.kt

```diff
@@ -30,196 +60,216 @@ private val THREAD_MESSAGES = createMessageList {
+
+        // should
+        stateList.last().apply {
+            this shouldBeInstanceOf MessageListViewModel.State.Result::class
+            val state = (this as MessageListViewModel.State.Result)
+            state.messageListItem.items.filterIsInstance<MessageListItem.MessageItem>().map {
+                it.message
+            } shouldBeEqualTo MESSAGES
+        }
+    }
+
+    @Test
+    fun `When delete event doesn't have hard flag Should delete message`() {
+        // given
+        val viewModel = MessageListViewModel(CID, chatClient = chatClient)
+        viewModel.state.observeAll()
+
+        // when
+        viewModel.onEvent(MessageListViewModel.Event.DeleteMessage(MESSAGE, hard = false))
+
+        // should
+        verify(chatClient).deleteMessage(MESSAGE.id, false)
+    }
+
+    @Test
+    fun `When delete event has hard flag Should hard delete message`() {
+        // given
+        val viewModel = MessageListViewModel(CID, chatClient = chatClient)
+        viewModel.state.observeAll()
+
+        // when
+        viewModel.onEvent(MessageListViewModel.Event.DeleteMessage(MESSAGE, hard = true))
+
+        // should
+        verify(chatClient).deleteMessage(MESSAGE.id, true)
+    }
+
+    @Test
+    fun `Should flag message`() {
+        // given
+        val viewModel = MessageListViewModel(CID, chatClient = chatClient)
+        viewModel.state.observeAll()
+
+        // when
+        viewModel.onEvent(MessageListViewModel.Event.FlagMessage(MESSAGE))
+
+        // should
+        verify(chatClient).flagMessage(MESSAGE.id)
+    }
+
+    @Test
+    fun `Should navigate up from normal mode on back click`() {
+        // given
+        val viewModel = MessageListViewModel(CID, chatClient = chatClient)
+        val states = viewModel.state.observeAll()
+
+        // when
+        viewModel.onEvent(MessageListViewModel.Event.BackButtonPressed)
+
+        // should
+        states.last() shouldBeEqualTo MessageListViewModel.State.NavigateUp
+    }
+
+    @Test
+    @Disabled(
+        "Can not be tested until we use mockk or other way to mock static function. " +
+            "ChatClient.getRepliesAsState is called."
+    )
+    fun `Should return from thread to normal mode on back click`() {
+        // given
+        val viewModel = MessageListViewModel(CID, chatClient = chatClient)
+        val states = viewModel.state.observeAll()
+
+        // when
+        viewModel.onEvent(MessageListViewModel.Event.ThreadModeEntered(MESSAGE))
+        viewModel.onEvent(MessageListViewModel.Event.BackButtonPressed)
+
+        // should
+        states.last().run {
+            this shouldBeInstanceOf MessageListViewModel.State.Result::class
+            (this as MessageListViewModel.State.Result).let { it ->
+                it.messageListItem.run {
+                    isThread shouldBeEqualTo false
+                    isTyping shouldBeEqualTo false
+
+                    val messages = items.run {
+                        get(1).apply {
```

Comment:
Can we please avoid using `run` as much as we have here? there are multiple context changes and scoping things going on here, where we have:

`run` for states.last, changing the `this` receiver to one thing, then having `it.messageListItem.run` to another scope, then using `items.run` and finally `apply` in several scenarios.

The complexity of this is through the roof and I'm fairly certain this should either be multiple tests or can be smaller functions that check each piece of behavior specifically or something.

Especially when you take a look at the test `Should return from thread to normal mode on back click`. How does navigating up or moving from thread to normal mode have anything to do with message positions in a given group?